File: ai-agent-station-study-infrastructure/src/main/java/cn/bugstack/ai/infrastructure/dao/po/AiClientApi.java
package cn.bugstack.ai.infrastructure.dao.po;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * AI客户端API配置表
 * @author bugstack虫洞栈
 * @description AI客户端API配置表 PO 对象
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AiClientApi {

    /**
     * 主键ID
     */
    private Long id;

    /**
     * API ID
     */
    private String apiId;

    /**
     * 基础URL
     */
    private String baseUrl;

    /**
     * API密钥
     */
    private String apiKey;

    /**
     * 对话补全路径
     */
    private String completionsPath;

    /**
     * 嵌入向量路径
     */
    private String embeddingsPath;

    /**
     * 状态(0:禁用,1:启用)
     */
    private Integer status;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

}

File: zhishu-ui/src/api/ai-client-api.ts
/**
 * 客户端 API 管理（/api/v1/admin/ai-client-api）
 * ⚠️ 注意：本模块的 update-by-id 后端为 **PUT**（旧 React 实现误用了 POST，会 405）。
 */
import { http } from './http'
import { ADMIN_BASE, CRUD } from './endpoints'

export interface AiClientApiItem {
  id: number
  apiId: string
  baseUrl: string
  apiKey?: string
  completionsPath?: string
  embeddingsPath?: string
  status: number
  createTime?: string
  updateTime?: string
}

const BASE = ADMIN_BASE.AI_CLIENT_API

export const AiClientApiApi = {
  queryList: (payload: Record<string, unknown>) => http.post<AiClientApiItem[]>(CRUD.queryList(BASE), payload),
  create: (payload: Record<string, unknown>) => http.post<boolean>(CRUD.create(BASE), payload),
  updateById: (payload: Record<string, unknown>) => http.put<boolean>(CRUD.updateById(BASE), payload),
  deleteById: (id: number | string) => http.delete<boolean>(CRUD.deleteById(BASE, id)),
  queryAll: () => http.get<AiClientApiItem[]>(CRUD.queryAll(BASE)),
  queryEnabled: () => http.get<AiClientApiItem[]>(CRUD.queryEnabled(BASE)),
}
